tests/unit/packaging/test_views.py

```diff
@@ -323,6 +325,134 @@ def test_long_singleline_license(self, db_request):
             "characters, it's really so lo..."
         )
 
+    @pytest.mark.parametrize(
+        ("base_url", "reference", "expected"),
+        [
+            (
+                "https://github.com/pypi/warehouse",
+                "2ec275dc2a05cd8ed2f0a9a0adadfcff9096d982",
+                (
+                    "https://github.com/pypi/warehouse/tree/"
+                    "2ec275dc2a05cd8ed2f0a9a0adadfcff9096d982"
+                ),
+            ),
+            (
+                "https://gitlab.com/sample/sample",
+                "refs/heads/main",
+                "https://gitlab.com/sample/sample/tree/refs/heads/main",
+            ),
+        ],
+    )
+    def test_format_url(self, base_url, reference, expected):
+        assert views.format_url(base_url, reference) == expected
+
+    @pytest.mark.parametrize(
+        ("publisher_name", "claims"),
+        [
+            (
+                "GitLab",
+                {
+                    "1.3.6.1.4.1.57264.1.8": "https://gitlab.com",
+                    "1.3.6.1.4.1.57264.1.9": (
+                        "https://gitlab.com/facutuesca/gitlab-oidc-project//"
```

Comment:
This uses @facutuesca GitLab project for a quick iteration but will be changed once a PR similar to this one will be merged https://github.com/pypi/warehouse/pull/17051